lib/portage/dbapi/vartree.py

```diff
@@ -1007,22 +1009,34 @@ def unpack_contents(self, pkg, dest_dir,
 		if include_unmodified_config is not None:
 			opts_list.append('--include-unmodified-config={}'.format(
 				'y' if include_unmodified_config else 'n'))
-
 		opts, args = parser.parse_known_args(opts_list)
 
-		tar_cmd = ('tar', '-x', '--xattrs', '--xattrs-include=*', '-C', dest_dir)
-		pr, pw = os.pipe()
-		proc = (yield asyncio.create_subprocess_exec(*tar_cmd, stdin=pr))
-		os.close(pr)
-		with os.fdopen(pw, 'wb', 0) as pw_file:
+		binpkg_format = settings.get("BINPKG_FORMAT", "xpak")
+		if binpkg_format == "xpak":
+			tar_cmd = ('tar', '-x', '--xattrs', '--xattrs-include=*', '-C', dest_dir)
+			pr, pw = os.pipe()
+			proc = (yield asyncio.create_subprocess_exec(*tar_cmd, stdin=pr))
+			os.close(pr)
+			with os.fdopen(pw, 'wb', 0) as pw_file:
+				excluded_config_files = (yield loop.run_in_executor(ForkExecutor(loop=loop),
+					functools.partial(self._dblink(cpv).quickpkg,
+					pw_file,
+					include_config=opts.include_config == 'y',
+					include_unmodified_config=opts.include_unmodified_config == 'y')))
+			yield proc.wait()
+			if proc.returncode != os.EX_OK:
+				raise PortageException('command failed: {}'.format(tar_cmd))
+		elif binpkg_format == "gpkg":
+			_, gpkg_tmp = tempfile.mkstemp(suffix=".gpkg.tar")
```

Comment:
The `_` variable here is an file descriptor that needs to be closed, or else we've got a file descriptor leak.